windows/Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM mcr.microsoft.com/windows/servercore:1809 AS caddy
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# https://github.com/caddyserver/dist/commits
+ENV CADDY_DIST_COMMIT 80870b227ded910971ecace4a0c136bf0ef46342
+
+# RUN set -eux; \
+# 	mkdir -p \
+# 		/config/caddy \
+# 		/data/caddy \
+# 		/etc/caddy \
+# 		/usr/share/caddy \
+# 	; \
+# 	wget -O /etc/caddy/Caddyfile "https://github.com/caddyserver/dist/raw/$CADDY_DIST_COMMIT/config/Caddyfile"; \
+# 	wget -O /usr/share/caddy/index.html "https://github.com/caddyserver/dist/raw/$CADDY_DIST_COMMIT/welcome/index.html"
+
+# https://github.com/caddyserver/caddy/releases
+ENV CADDY_VERSION v2.0.0-rc.3
+
+RUN Invoke-WebRequest \
+        -Uri "https://github.com/caddyserver/caddy/releases/download/v2.0.0-rc.3/caddy_2.0.0-rc.3_windows_amd64.zip" \
+        -OutFile "/caddy.zip"; \
+    Expand-Archive -Path "/caddy.zip" -DestinationPath "/" -Force; \
+    Remove-Item "/caddy.zip" -Force
+
+# See https://caddyserver.com/docs/conventions#file-locations for details
+# ENV XDG_CONFIG_HOME=/config
+# ENV XDG_DATA_HOME=/data
+
+VOLUME /config
+VOLUME /data
+
+LABEL org.opencontainers.image.version=v2.0.0-rc.3
+LABEL org.opencontainers.image.title=Caddy
+LABEL org.opencontainers.image.description="a powerful, enterprise-ready, open source web server with automatic HTTPS written in Go"
+LABEL org.opencontainers.image.url=https://caddyserver.com
+LABEL org.opencontainers.image.documentation=https://caddyserver.com/docs
+LABEL org.opencontainers.image.vendor="Light Code Labs"
+LABEL org.opencontainers.image.licenses=Apache-2.0
+LABEL org.opencontainers.image.source="https://github.com/caddyserver/caddy-docker"
+
+EXPOSE 80
+EXPOSE 443
+EXPOSE 2019
+
+CMD ["caddy", "run", "--config", "/etc/caddy/Caddyfile", "--adapter", "caddyfile"]
```

Comment:
FWIW, `--adapter caddyfile` isn't necessary when the `--config` filename starts with `Caddyfile`.

Reply:
Doesn't hurt to be explicit though IMO 🤷‍♂️

Reply:
It just means that if the user doesn't use a Caddyfile and wants to use JSON, they have to remember to remove the `--adapter` flag, even though they're using Caddy's native format. I don't care too much either way, just wanted to point this out. I don't want people thinking they're required to use that flag for the Caddyfile all the time.

Reply:
When you run with docker, `CMD` is just the default command. If you specify a different one, it just overrides the existing one entirely. The user wouldn't need to remove anything if they run with JSON.

e.g. `docker run caddy caddy run --config /etc/caddy/config.json`

Reply:
This is likely to change too - the image is very much not working in Windows right now 😉